test/EFCore.PG.FunctionalTests/MigrationsNpgsqlTest.cs

```diff
@@ -1,295 +1,2074 @@
                 });
 
-                modelBuilder.Entity("ModelSnapshot22.Post", b =>
+            AssertSql();
+        }
+
+        [Fact]
+        public virtual async Task Add_column_with_huge_varchar()
+        {
+             // PostgreSQL doesn't allow varchar(x) with x > 10485760, so we map this to text.
+             // See #342 and https://www.postgresql.org/message-id/15790.1291824247%40sss.pgh.pa.us
+            await Test(
+                builder => builder.Entity(
+                    "People", e =>
+                    {
+                        e.Property<int>("Id");
+                        e.HasKey("Id");
+                    }),
+                builder => { },
+                builder => builder.Entity("People").Property<string>("Name").HasMaxLength(10485761),
+                model =>
                 {
-                    b.HasOne("ModelSnapshot22.Blog", "Blog")
-                        .WithMany("Posts")
-                        .HasForeignKey("BlogId");
+                    var table = Assert.Single(model.Tables);
+                    var column = Assert.Single(table.Columns, c => c.Name == "Name");
+                    Assert.Equal("text", column.StoreType);
                 });
-#pragma warning restore 612, 618
+
+            AssertSql(
+                @"ALTER TABLE ""People"" ADD ""Name"" text NULL;");
+        }
+
+        public override async Task Alter_column_change_type()
+        {
+            await base.Alter_column_change_type();
+
+            AssertSql(
+                @"ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" TYPE bigint;
+ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" SET NOT NULL;
+ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" DROP DEFAULT;");
+        }
+
+        public override async Task Alter_column_make_required()
+        {
+            await base.Alter_column_make_required();
+
+            AssertSql(
+                @"ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" TYPE text;
+ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" SET NOT NULL;
+ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" DROP DEFAULT;");
+        }
+
+        public override async Task Alter_column_make_required_with_index()
+        {
+            await base.Alter_column_make_required_with_index();
+
+            AssertSql(
+                @"ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" TYPE text;
+ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" SET NOT NULL;
+ALTER TABLE ""People"" ALTER COLUMN ""SomeColumn"" DROP DEFAULT;");
+        }
+
+        public override async Task Alter_column_make_required_with_composite_index()
+        {
+            await base.Alter_column_make_required_with_composite_index();
+
+            AssertSql(
+                @"ALTER TABLE ""People"" ALTER COLUMN ""FirstName"" TYPE text;
+ALTER TABLE ""People"" ALTER COLUMN ""FirstName"" SET NOT NULL;
+ALTER TABLE ""People"" ALTER COLUMN ""FirstName"" DROP DEFAULT;");
```

Comment:
Unless I'm mistaken, when you see multiple strings with a comment separating them, this means the migrations generator is generating multiple ADO commands. In other cases (like here), multiple statements are batched in the same command. In general it doesn't make much of a difference (we don't care about network roundtrips in migrations), but it could be an issue where we need to suppress transactions, since that holds for the entire command.